pennylane/templates/subroutines/arithmetic/semi_adder.py

```diff
@@ -56,6 +61,29 @@ def _right_operator(wires, ik_is_zero=False):
     return [CNOT([ck, aux]), adjoint(TemporaryAND([ck, tk, aux]))]
 
 
+def _controlled_right_operator(wires, control_wires, control_values, ik_is_zero=False):
```

Comment:
What is `ik_is_zero`? (Might want to expand the docstring to explain the arguments.)